gpu/matrix/dense_kernels.cpp

```diff
@@ -1,266 +0,0 @@
-/*******************************<GINKGO LICENSE>******************************
```

Comment:
Note that this file was just renamed to `dense_kernels.cu`.
Other than that, the changes in it are the following:

*   the kernels have been changed to throw a NotImplemented exception in case the value type is not supported by cublas
*   an implementation of the `compute_dot` kernel for datatypes not supported by cuBLAS has been  added